bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -173,9 +173,11 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 	log = log.WithValues(configOwner.GetKind(), klog.KRef(configOwner.GetNamespace(), configOwner.GetName()), "resourceVersion", configOwner.GetResourceVersion())
 
-	log = log.WithValues("Cluster", klog.KRef(configOwner.GetNamespace(), configOwner.ClusterName()))
+	log = log.WithValues("Cluster", klog.KRef(configOwner.GetNamespace(), configOwner.ClusterName()), "KubeadmConfig", klog.KRef(config.GetNamespace(), config.GetName()))
```

Comment:
You mean `KubeadmConfig` which I added? I definitely don't see it in a failing test's log, so that's why I wanted to have it.

Example _before_:

```
I0817 15:10:42.545337   55146 kubeadmconfig_controller.go:179]  "msg"="Reconciling KubeadmConfig" "Cluster"={"name":"cluster","namespace":"default"} "MachinePool"={"name":"worker-machinepool","namespace":"default"} "resourceVersion"="999"
```

I found controller-runtime to actually fill it for production controller logs (namely in `func (blder *Builder) doController(r reconcile.Reconciler) error`):

```
kubeadmconfig_controller.go:314] "Refreshing token until the infrastructure has a chance to consume it" controller="kubeadmconfig" controllerGroup="bootstrap.cluster.x-k8s.io" controllerKind="KubeadmConfig" KubeadmConfig="mynamespace/my-nodepool0"
```

So my problem is that tests never use the controller-runtime builder but construct `KubeadmConfigReconciler{...}` directly. Not worth fixing this as I did, so I'll revert. Overall though, having the object logged in the test logs would help the developer experience.

Reply:
Didn't look at the code but potentially we could try to use the builder in tests as well

Reply:
Hm probably more complicated then that

Reply:
That would fix it - currently most of the KCP tests set up the controller using the raw reconciler e.g. 

```go
		r := &KubeadmControlPlaneReconciler{
```

Not sure it's worth it for tests where you can set things up to get info on which cluster etc. is involved in the test code. But would be fine with it if someone wanted to make the changes